### PR TITLE
osd: Pointer initialization after delete

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -8676,6 +8691,7 @@ void OSD::handle_pg_backfill_reserve(OpRequestRef op)
 	m->query_epoch,
 	PG::RemoteReservationRevoked()));
   } else {
+    delete m;
```
Comment (liewegas) [inline src/osd/OSD.cc]: you can drop these delete calls.. we're about to abort so they just make the code more verbose

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
@liewegas please check if you get a chance.

File: src/osd/OSD.cc [hunk 2]
```diff
@@ -2047,13 +2049,19 @@ OSD::OSD(CephContext *cct_, ObjectStore *store_,
 OSD::~OSD()
 {
   delete authorize_handler_cluster_registry;
+  authorize_handler_cluster_registry = nullptr;
```
Comment (tchaikov) [inline src/osd/OSD.cc]: what is the point of setting a member variable to null here? as i think it is a dtor?

File: src/osd/OSD.cc [hunk 3]
```diff
@@ -8730,6 +8740,7 @@ void OSD::handle_pg_recovery_reserve(OpRequestRef op)
 	m->query_epoch,
 	PG::DeferRecovery(0.0)));
   } else {
+    delete m;
```
Comment (liewegas) [inline src/osd/OSD.cc]: here too

Comment (shinobu-x) [reply src/osd/OSD.cc]: Nice comeback!

File: src/osd/OSD.cc (hunk 2 above)
Comment (tchaikov) [reply src/osd/OSD.cc]: @shinobu-x this is not addressed.

**Comment by shinobu-x** [comment]
@tchaikov like now?

**Comment by tchaikov** [comment]
@shinobu-x https://github.com/ceph/ceph/pull/19389/files#r156918110

**Comment by shinobu-x** [comment]
@tchaikov should be fine?

**Comment by shinobu-x** [comment]
test this please

File: src/osd/OSD.cc [hunk 4]
```diff
@@ -2052,8 +2054,11 @@ OSD::~OSD()
   cct->get_perfcounters_collection()->remove(recoverystate_perf);
   cct->get_perfcounters_collection()->remove(logger);
   delete recoverystate_perf;
+  recoverystate_perf = nullptr;
```
Comment (tchaikov) [inline src/osd/OSD.cc]: https://github.com/ceph/ceph/pull/19389#discussion_r155913249

File: src/osd/OSD.cc [hunk 5]
```diff
@@ -2052,8 +2054,11 @@ OSD::~OSD()
...
   delete logger;
+  logger = nullptr;
```
Comment (tchaikov) [inline src/osd/OSD.cc]: https://github.com/ceph/ceph/pull/19389#discussion_r155913249

File: src/osd/OSD.cc [hunk 6]
```diff
@@ -2052,8 +2054,11 @@ OSD::~OSD()
...
   delete store;
+  store = nullptr;
```
Comment (tchaikov) [inline src/osd/OSD.cc]: https://github.com/ceph/ceph/pull/19389#discussion_r155913249

**Comment by jdurgin** [comment]
Closing since this sort of clean up would need a rebase.

**Pull request closed without merging**